src/app/screens/Liquid/ConfirmSignPset.tsx

```diff
@@ -0,0 +1,219 @@
+  EsploraAssetRegistry,
+  LiquidAddress,
+  OriginData,
+  PsetPreview,
+} from "~/types";
+
+function ConfirmSignPset() {
+  const navState = useNavigationState();
+  const { t: tCommon } = useTranslation("common");
+  const { t } = useTranslation("translation", {
+    keyPrefix: "confirm_sign_pset",
+  });
+  const navigate = useNavigate();
+
+  const pset = navState.args?.pset;
+  const origin = navState.origin as OriginData;
+  const [loading, setLoading] = useState(false);
+  const [successMessage, setSuccessMessage] = useState("");
+  const [preview, setPreview] = useState<PsetPreview | undefined>(undefined);
+  const [assetRegistry, setAssetRegistry] = useState<EsploraAssetRegistry>({});
+  const [showAddresses, setShowAddresses] = useState(false);
+  const [showHex, setShowHex] = useState(false);
+
+  useEffect(() => {
+    (async () => {
+      if (!pset) throw new Error("pset is undefined");
+      const preview = await api.liquid.getPsetPreview(pset);
+      setPreview(preview);
+
+      const registry = await api.liquid.fetchAssetRegistry(preview);
+      setAssetRegistry(registry);
+    })();
+  }, [pset]);
+
+  async function confirm() {
+    try {
+      if (!pset) throw new Error("pset is undefined");
+      setLoading(true);
+      const response = await api.liquid.signPset(pset);
+
+      msg.reply(response);
+      setSuccessMessage(tCommon("success"));
+    } catch (e) {
+      console.error(e);
+      if (e instanceof Error) toast.error(`${tCommon("error")}: ${e.message}`);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  function reject(e: React.MouseEvent<HTMLAnchorElement>) {
+    e.preventDefault();
+    msg.error(USER_REJECTED_ERROR);
+  }
+
+  function close(e: React.MouseEvent<HTMLButtonElement>) {
+    if (navState.isPrompt) {
+      window.close();
+    } else {
+      e.preventDefault();
+      navigate(-1);
+    }
+  }
+
+  function toggleShowAddresses() {
+    setShowAddresses((current) => !current);
+  }
+  function toggleShowHex() {
+    setShowHex((current) => !current);
+  }
+
+  if (!preview) {
+    return <Loading />;
+  }
+
+  return (
+    <div className="h-full flex flex-col overflow-y-auto no-scrollbar">
+      <ScreenHeader title={t("title")} />
+      {!successMessage ? (
+        <Container justifyBetween maxWidth="sm">
+          <div className="flex flex-col gap-4 mb-4">
+            <PublisherCard
+              title={origin.name}
+              image={origin.icon}
+              url={origin.host}
+            />
+            <div className="p-4 shadow bg-white dark:bg-surface-02dp rounded-lg overflow-hidden flex flex-col gap-4">
+              <div className="flex justify-between items-start">
+                <h2 className="font-medium dark:text-white">
+                  {t("allow_sign", { host: origin.host })}{" "}
+                </h2>
+                <Hyperlink
+                  href="https://medium.com/blockstream/pset-constructing-confidential-transactions-437109bb4ac5"
```

Comment:
we should link to something we control. So let's make an Alby guide page for this - and from there link to such resources.

Reply:
I made a page here. Needs some more work: https://guides.getalby.com/overall-guide/alby-browser-extension/features/liquid